pytential/collection.py

```diff
@@ -170,47 +170,47 @@ def __init__(self,
             assert isinstance(places, Mapping)
             places_dict = places
 
-        self.places = immutables.Map(places_dict)
-        self.auto_where = (auto_source, auto_target)
-
-        self._caches: dict[str, Any] = {}
-
```

Comment:
This is just a bit of churn that moves all the `self.attr = value` after the validation was done.